src/fmt.ml

```diff
@@ -94,6 +103,20 @@ let doomed ppf reason =
   pf ppf "Something@ unreasonable@ is@ going@ on (%a).@ You@ are@ doomed."
     text reason
 
+(* Concatenation *)
+
+let concat (pp1 : 'a t) (pp2 : 'b t) ppf (v1, v2) =
+  pp1 ppf v1 ; pp2 ppf v2
+
+let prefix pp_p pp_v ppf v = concat pp_p pp_v ppf ((),v)
+
+let suffix pp_s pp_v ppf v = concat pp_v pp_s ppf (v,())
+
+let (<>) = concat
```

Comment:
I'm little bit wary of operators of general and we'll see if I keep them. But that one is out of question. Pervasives.(<>) should not be overriden that's a code reading nightmare; did you mean  `<.>` ?